### DISCUSSION TITLE
How to change the scrolling speed when using keyboard?

### DISCUSSION THREAD

**Discussion by phanen** [sumatrapdfreader/sumatrapdf#2458] [Q&A] [answered]
I prefer to using the keyboard for page scrolling.
But using `j/k` or `↑/↓` is too slow. 
Is there any way to change the speed of the scrolling?

**Comment by GitHubRulesOK** [comment, marked as answer]
Every machine and zoom factor has an effect, (you could even have system keyboard driver controls) however,
You can use the ctrl or shift keys to accelerate steps,
but even in my machine non accelerated up down is often too fast within a page.

There is also in latest pre-release the optional `SmoothScroll = false` which may also have a per usage effect

**Reply by phanen** [reply]
Thanks for advice.

I find that the ahk script can improve the speed better, though the effect of long press will still be delayed. 
```autohotkey 
    ; Scroll up
    CapsLock & a::Send,{WheelUp}
    ; Scroll down    
    CapsLock & s::Send,{WheelDown}
```
Maybe I just want the page to scroll more smoothly (like `j/k` in vimium).

**Comment by kjk** [comment]
The short answer is: there is no way to change that.

**Comment by kjk** [comment]
In next [pre-release](https://www.sumatrapdfreader.org/prerelease) (16478 or later) you can create a [custom keyboard shortcut](https://www.sumatrapdfreader.org/docs/Customizing-keyboard-shortcuts) and bind it to `CmdScrollUp <n>` and `CmdScrollDown <n>` commands.
For example, to scroll by 5 lines:
```
Shortcuts [
	[
		Cmd = CmdScrollUp 5
		Key = j
	]
	[
		Cmd = CmdScrollDown 5
		Key = k
	]
]
```
Or you can scroll by 10 lines or whatever and bind it to any key you want.
